src/main/java/com/desiato/puresynth/services/CustomAuthenticationService.java

```diff
@@ -0,0 +1,35 @@
+package com.desiato.puresynth.services;
+
+import com.desiato.puresynth.models.User;
+import com.desiato.puresynth.repositories.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Service
+public class CustomAuthenticationService {
+
+    private static final Logger logger = LoggerFactory.getLogger(CustomAuthenticationService.class);
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+    public boolean authenticate(String email, String password) {
+        User user = userRepository.findByEmail(email);
```

Comment:
Can you change `userRepository.findByEmail` to return `Optional<User>? Optionals are better and safer than nulls.